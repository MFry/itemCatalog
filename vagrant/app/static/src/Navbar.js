import React from 'react';
import logo from './logo.svg';


const Navbar = () => (
  <nav>
    <div className="nav-wrapper container">
      <a href="/" className="brand-logo"><img src={logo} className="App-logo" alt="logo" /></a>
      <ul id="nav-mobile" className="right">
        <li><button className="waves-effect waves-light btn">Login</button></li>
      </ul>
    </div>
  </nav>
);
export default Navbar;
