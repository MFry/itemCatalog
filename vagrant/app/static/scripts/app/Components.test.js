import React from 'react';
import { shallow } from 'enzyme';
import serializer from 'enzyme-to-json/serializer';
import Navbar from './Navbar';
import Header from './Header';

expect.addSnapshotSerializer(serializer);

describe('Components', () => {
    describe('Navbar', () => {
        const navbar = shallow(<Navbar />);
        it('Should render', () => {
            expect(navbar.find('nav').hasClass('navbar')).toBe(true);
        });
        it('Should contain brand text', () => {
            expect(navbar.findWhere(n => n.is('.navbar-brand')).text()).toBe(
                'Catalog App'
            );
        });
        it('Should match snapshot', () => {
            expect(navbar).toMatchSnapshot();
        });
    });
    describe('Header', () => {
        const header = shallow(<Header />);
        it('Should render', () => {
            expect(header.find('.page-header').type('div')).toBe('div');
        });
        it('Should contain Catalog text', () => {
            expect(header.findWhere(n => n.is('.col-md-4')).text()).toBe(
                'Catalog App'
            );
        });
        it('Should match snapshot', () => {
            expect(header).toMatchSnapshot();
        });
    });
});
