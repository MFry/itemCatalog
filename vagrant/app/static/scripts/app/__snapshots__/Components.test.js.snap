// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Header Should match snapshot 1`] = `
<div
  className="page-header"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-4"
      >
        <h2>
          Catalog App
        </h2>
      </div>
    </div>
  </div>
</div>
`;

exports[`Components Navbar Should match snapshot 1`] = `
<nav
  className="navbar navbar-default navbar-fixed-top"
>
  <div
    className="container-fluid"
  >
    <div
      className="col-md-10 navbar-header"
    >
      <div
        className="navbar-brand"
      >
        Catalog App
      </div>
    </div>
    <div
      className="col-md-2"
    >
      <button
        className="btn btn-default navbar-btn"
        type="button"
      >
        Sign in
      </button>
    </div>
  </div>
</nav>
`;
